pallets/api/src/fungibles/mod.rs

```diff
@@ -150,11 +192,11 @@ pub mod pallet {
 			spender: AccountIdOf<T>,
 			value: BalanceOf<T>,
 		) -> DispatchResultWithPostInfo {
-			let who = ensure_signed(origin.clone())
+			let owner = ensure_signed(origin.clone())
 				.map_err(|e| e.with_weight(Self::weight_approve(0, 0)))?;
-			let current_allowance = AssetsOf::<T>::allowance(id.clone(), &who, &spender);
-			let spender = T::Lookup::unlookup(spender);
-			let id: AssetIdParameterOf<T> = id.into();
+			let current_allowance = AssetsOf::<T>::allowance(id.clone(), &owner, &spender);
+			let spender_unlookup = T::Lookup::unlookup(spender.clone());
+			let id_param: AssetIdParameterOf<T> = id.clone().into();
```

Comment:
`id_param` is not used when `Equal`, better to move it into the `Greater`/`Less` match arms to avoid the unnecessary clone when equal. Very low likelihood of the code path being run but an easy optimisation.

In the Greater case, it can be inserted directly inline into line 209 without the need for the variable.

Reply:
see commit `refactor: optimization pallet fungibles`